src/main/java/Deadline.java

```diff
@@ -1,14 +1,64 @@
 //Deadline is one of 3 types of Task, so it must be a child of Task
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 
 public class Deadline extends Task {
 
-    protected String dateline;
+    private final Date byWhen;
+    private final String dateline;
 
     public Deadline(String name, String dateline) {
         super(name);
         this.dateline = dateline;
+        this.byWhen = convertStringIntoDateFormat(dateline);
     }
 
+    /**
+     * Converts a String input given by the user into a Date of a given format.
+     * @param dateline String of date of deadline given by user.
+     * @return A Date object with the same details as what the user has given.
+     */
+    public Date convertStringIntoDateFormat(String dateline) {
```

Comment:
I get it that you are trying to convert the string input of the date into a Date object, then converting it back into a string format for printing, perhaps these 2 methods can be combined, which can make your code neater since lesser methods are being called as well.